Meson bump

AndersonTorres opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Get rid of old Meson

https://github.com/NixOS/nixpkgs/pull/144779

As I have said before, the plan is:

- Insert new Meson in parallel with old one, the older being a safe default (done)
- Try to force new Meson as default
- Look at the ofborg logs in order to repair those packages that do not accept new Meson yet
  - Updates should be preferrable

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

AndersonTorres committed:
meson: change default to 0.60

AndersonTorres committed:
metadata-cleaner: change meson attribute

Meson 0.60 is the default now.

AndersonTorres committed:
libadwaita: change meson attribute

Meson 0.60 is the default now.

AndersonTorres committed:
systemd: use old Meson

Systemd fails to build with new Meson. According to Ofborg logs at

https://gist.github.com/GrahamcOfBorg/d1304f0fc0ed308713303361273522a7

```
meson.build:3:0: ERROR: Unknown options: "cryptsetup, kill-path"
```

primeos commented on pkgs/development/libraries/wlroots/0.15.nix:
```diff
@@ -1,7 +1,28 @@
-, libGL, wayland, wayland-protocols, libinput, libxkbcommon, pixman
-, xcbutilwm, libX11, libcap, xcbutilimage, xcbutilerrors, mesa
-, libpng, ffmpeg, xcbutilrenderutil, seatd, vulkan-loader, glslang
+{ lib
+, stdenv
+, fetchFromGitLab
+, ffmpeg
+, glslang
+, libGL
+, libX11
+, libcap
+, libinput
+, libpng
+, libxkbcommon
+, mesa
+, meson
+, ninja
+, pixman
+, pkg-config
+, seatd
+, vulkan-loader
+, wayland
+, wayland-protocols
+, wayland-scanner
+, xcbutilerrors
+, xcbutilimage
+, xcbutilrenderutil
+, xcbutilwm
```
Not a blocker but I hate those refactorings in unrelated commits (it's a bad commit practice). We should either reformat all of Nixpkgs at once or packages individually. A commit like https://github.com/NixOS/nixpkgs/pull/153082/commits/3d74e5725990dee4b934a8a1be8f8aa615c5da9e ("wlroots 0.15: change meson attribute - Meson 0.60 is the default now") shouldn't contain such unrelated changes.

AndersonTorres replied:
Understandable.

I prefer to do such code refactorings among other things because they did not change the semantics of the code.

AndersonTorres committed:
wlroots 0.15: refactor

A quick and easy rewriting.

AndersonTorres committed:
wlroots 0.15: change meson attribute

Meson 0.60 is the default now.

AndersonTorres committed:
wlroots 0.15: change meta.platforms

wlroots runs whenever wayland runs. Nowadays Wayland is linux-only, however
there are some projects attempting to port it to other Unices. Therefore, it is
better to bind meta.platforms to Wayland.

AndersonTorres committed:
GTK 4.x: change meson attribute

Meson 0.60 is the default now.
